### PR TITLE
Remove CMake < 3 CMP0042 workarounds

### PR DESCRIPTION
**Pull request by haampie**
**Description**

TL;DR: This deletes obsolete CMake code and improves upon #1002
by sticking to (sensible) CMake defaults, in accordance to the principle
of least astonishment ;)

CMake only sets install names on darwin to `@rpath/<name>` (which is
desired otherwise rpaths don't work at all) when CMP0042 is ON. That's
the default when CMake 3.0 or higher is required. And lapack requires it
already for years: as of v3.9.1 (8f004b353a876a117b9e9428fff3becdd22c79ce).

So, delete the old workarounds that effectively set CMP0042 to ON.

Further, delete the following three options that are redefinitions of
builtins and have values that are default anyways:
- `CMAKE_MACOSX_RPATH`
- `CMAKE_SKIP_BUILD_RPATH`
- `CMAKE_BUILD_WITH_INSTALL_RPATH`

Lastly, lapack automatically sets `CMAKE_INSTALL_RPATH_USE_LINK_PATH` to
`ON` whenever installing to a non-system dir. The assumption is that
whenever you install something to a non-system dir, you need rpaths to
locate dependencies. But this is just an assumption which may or may
not hold. The downside of it is that the option can be annoying when
lapack is used as a sub-project as it affects a global CMake variable
(for example OpenBLAS uses lapack as a sub-project). Instead, let users
or packagers provide this on the command line if they really need it --
remove it from lapack as it's as helpful as it is harmful.

**Checklist**
- [ ] The documentation has been updated.
- [ ] If the PR solves a specific issue, it is set to be closed on merge.

### CODE REVIEW

**Comment by haampie** [comment]
ping @langou / @ahnaf-tahmid-chowdhury (ref #1002)

**Review by ahnaf-tahmid-chowdhury — APPROVED** [review]
I agree. Leveraging CMake's built-in capabilities whenever possible is generally a good approach.

**Review by langou — APPROVED** [review]
